react/FilterNavigator.js

```diff
@@ -66,6 +73,101 @@ class FilterNavigator extends Component {
     hiddenFacets: {},
   }
 
+  state = {
+    openContent: false,
+    filtersChecks: {},
+    pagesArgs: {},
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const nonEmptyFilters = this.mapCheckboxFilters(this.filters.filter(spec => spec.options.length > 0))
+
+    if (this.willUpdateFilters(Object.keys(nonEmptyFilters), Object.keys(prevState.filtersChecks))) {
+      this.setState({ filtersChecks: nonEmptyFilters })
+    }
+  }
+
+  willUpdateFilters = (newFilters, prevFilters) => {
+    if (newFilters.length !== prevFilters.length) return true
+    return !newFilters.every(e => prevFilters.includes(e))
+  }
+
+  mapCheckboxFilters = (filters) => {
+    const { map, rest } = this.props
+
+    const mapperOption = (isSelected, selectedFilters) => opt => {
+      const pagesArgs = formatFacetToLinkPropsParam(opt.type, opt)
+      return {
+        name: opt.Name,
+        ...pagesArgs,
+        checked: isSelected || !!find(propEq('Name', opt.Name), selectedFilters)
+      }
+    }
+    const mapperOptionsForFilter = (map, rest) => ({ options, type }) => mountOptions(options, type, map, rest).map(mapperOption(false, this.selectedFilters))
+    const reducerForOptions = (current, next) => union(current, next)
+    
+    let filtersAll = [...(filters.map(mapperOptionsForFilter(map, rest)).reduce(reducerForOptions, [])), ...this.selectedFilters.map(mapperOption(true))]
+    const finalFilters = uniqBy(pick(['name']), filtersAll)
+
+    return mergeAll(finalFilters.map(({ name, ...rest }) => ({ [name]: rest })))
+  }
+
+  handleFilterCheck = (filter) => {
+    const { getLinkProps } = this.props
+    const filterCopy = { ...this.state.filtersChecks }
+    filterCopy[filter].checked = !filterCopy[filter].checked
+
+    this.setState({
+      filtersChecks: filterCopy
+    })
+
+    const checkedFilters = []
+    for (const filter in filterCopy) {
+      if (filterCopy[filter].checked) {
+        checkedFilters.push(filterCopy[filter])
+      }
+    }
+
+    const map = checkedFilters.map(opt => { return getMapByType(opt.type) })
+    map.unshift('ft')
+    const pagesArgs = getLinkProps(checkedFilters, false, checkedFilters.map(opt => { return opt.slug }), map)
+    this.setState({ pagesArgs })
+  }
+
+  handleUpdateContentVisibility = () => {
+    this.setState({
+      openContent: false,
+    })
+  }
+
+  handleClickButton = event => {
+    if (!this.props.hideContent) {
+      this.setState({
+        openContent: !this.state.openContent,
+      })
+    }
+    event.persist()
+  }
+
+  clearFilters = () => {
+    const { getLinkProps, runtime: { navigate } } = this.props
+    const clearedFilters = { ...this.state.filtersChecks }
+    for (const filter in clearedFilters) {
```

Comment:
Use `map`instead

Reply:
Can't do, clearedFilters is an json Object.